File: Selu383.SP25.P03.Api/Features/Movies/Movie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Selu383.SP25.P03.Api.Features.Movies
{
    public class Movie
    {
    public int Id { get; set; }
    
    [Required]
    public required string Title { get; set; }
    
    public string? Description { get; set; }
    
    public string? PosterUrl { get; set; }
    
    public int Runtime { get; set; }
    
    public string? Rating { get; set; }
    
    public DateTime ReleaseDate { get; set; }
    
    public int? TmdbId { get; set; }
    
    public virtual ICollection<Showtime> Showtimes { get; set; } = new List<Showtime>();
    }
}

File: Selu383.SP25.P03.Mobile/types/models/movie.ts
export interface Movie {
  id: number;
  title: string;
  posterUrl: string;
  runtime: number;
  description?: string;
  rating?: string;
  releaseDate?: Date;
  tmdbId?: number;
  genre?: string;
  trailerUrl?: string;
}

export interface Showtime {
  id: number;
  startTime: string;
  ticketPrice: number;
  movieId: number;
  movieTitle: string;
  screenId: number;
  screenName: string;
  theaterId: number;
  theaterName: string;
}

export interface ShowtimesByTheater {
  theaterId: number;
  theaterName: string;
  distance?: string;
  showtimes: {
    id: number;
    startTime: string;
  }[];
}
